duckduckgo/privacy-essentials-safari
# ad click attribution follow up

Task/Issue URL: https://app.asana.com/0/1148434556963201/1202833154365438/f
Tech Design URL:
CC:

**Description**:

* [Remove support for non-ad_domain parameters](https://app.asana.com/0/0/1202676972107650/f)
* [Pixels to expand post-release](https://app.asana.com/0/0/1202639551643286/f)
* Fixes a memory leak that appeared to be causing a crash

**Steps to test this PR**:

Note that a DEBUG build changes some things to make this easier to test:

a. The interval validating a heuristic ad click validation is increased from 0.3 seconds to 5.0 seconds
b. The page loads pixel fires every 60 seconds rather than 24 hours.

***Test Case 1 Heuristic Validation Matching***
1. Open Console.app and filter on `ValidationTimer` and separately `Pixel fired`
1. Search for something to buy and click on an ad link without the ad_domain parameter (shopping tab usually has these)
3. Click on a few pages on the same site within 5 seconds of each page load
5. Console will show the ValidationTimer lifecycle
7. After 5 seconds a pixel will be fired `m_ad_click_heuristic_validation_1`

***Test Case 2 Heuristic Validation Mismatching***
1. Open Console.app and filter on `ValidationTimer` and separately `Pixel fired`
1. Search for something to buy and click on an ad link without the ad_domain parameter (shopping tab usually has these)
3. Immediately open a different website in the same tab
7. After 5 seconds a pixel will be fired `m_ad_click_heuristic_validation_0`

***Test Case 3 Ad click page loads***

1. Find a site that loads bat.js (e.g. "John Lewis" or "The China Shop")
3. Search for an ad that links to that site
5. Check the site has tried to load bat.js ("Third Party Resources screen")
6. Within 60 seconds, click around the site a bit
8. After 60 seconds you will see the `m_pageloads_with_ad_attribution` pixel fired with count parameter that should match the number of page loads.  Note that this will usually only fire every 24 hours.

***Test Case 4 Dashboard***

1. Smoke test the dashboard on a number of sites.


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
